client/me/memberships/subscription.jsx

```diff
@@ -11,114 +12,124 @@ import HeaderCake from 'calypso/components/header-cake';
 import { withLocalizedMoment } from 'calypso/components/localized-moment';
 import Main from 'calypso/components/main';
 import Notice from 'calypso/components/notice';
+import { CALYPSO_CONTACT } from 'calypso/lib/url/support';
 import titles from 'calypso/me/purchases/titles';
 import MeSidebarNavigation from 'calypso/me/sidebar-navigation';
 import { requestSubscriptionStop } from 'calypso/state/memberships/subscriptions/actions';
 import {
 	getSubscription,
 	getStoppingStatus,
 } from 'calypso/state/memberships/subscriptions/selectors';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
 import { purchasesRoot } from '../purchases/paths';
 import MembershipSiteHeader from './header';
 
 import './subscription.scss';
 
-class Subscription extends React.Component {
-	stopSubscription = () => this.props.requestSubscriptionStop( this.props.subscription.ID );
+function Subscription( { translate, subscription, moment, stoppingStatus } ) {
+	const dispatch = useDispatch();
 
-	render() {
-		const { translate, subscription, moment, stoppingStatus } = this.props;
+	const stopSubscription = () => dispatch( requestSubscriptionStop( subscription.ID ) );
 
-		return (
-			<Main wideLayout className="memberships__subscription">
-				<DocumentHead title={ translate( 'Subscription Details' ) } />
-				<MeSidebarNavigation />
-				<QueryMembershipsSubscriptions />
-				<FormattedHeader brandFont headerText={ titles.sectionTitle } align="left" />
-				<HeaderCake backHref={ purchasesRoot }>{ translate( 'Subscription Details' ) }</HeaderCake>
-				{ stoppingStatus === 'start' && (
-					<Notice
-						status="is-info"
-						isLoading={ true }
-						text={ translate( 'Stopping this subscription' ) }
-					/>
-				) }
-				{ stoppingStatus === 'fail' && (
-					<Notice
-						status="is-error"
-						text={ translate( 'There was a problem while stopping your subscription' ) }
-					/>
-				) }
-				{ stoppingStatus === 'success' && (
-					<Notice
-						status="is-success"
-						text={ translate( 'This subscription has been stopped. You will not be charged.' ) }
-					/>
-				) }
+	useEffect( () => {
+		if ( stoppingStatus === 'fail' ) {
+			// run is-error notice to contact support
+			dispatch(
+				errorNotice(
+					translate(
+						'There was a problem while stopping your subscription, please {{a}}{{strong}}contact support{{/strong}}{{/a}}.',
```

Comment:
This changes the translation, but since this is an error state it's probably not that common and I don't know if it's worth using `hasTranslation` (or a string freeze) in this case. I'll leave that decision to you.

Reply:
True, though I did add the string freeze tag already. Is it easy enough to just remove it? Or has a cascading series of events already started?

Reply:
I honestly don't know.